### PR TITLE
Add localization support for CMMN Engine

### PR DESCRIPTION
**Pull request by DavidPLamas**
#### Check List:
* Unit tests: YES
* Documentation: NO

### CODE REVIEW

File: modules/flowable-cmmn-api/src/main/java/org/flowable/cmmn/api/history/HistoricPlanItemInstanceQuery.java [hunk 1]
```diff
@@ -12,12 +12,12 @@
  */
 package org.flowable.cmmn.api.history;
 
+import org.flowable.common.engine.api.query.Query;
```
Comment (dbmalkovsky) [inline modules/flowable-cmmn-api/src/main/java/org/flowable/cmmn/api/history/HistoricPlanItemInstanceQuery.java:15]: Please note that project standards specify that imports for java classes are before those of `com` and `org` as was done originally in the file.  Thanks.

File: modules/flowable-cmmn-engine/src/main/java/org/flowable/cmmn/engine/CaseDefinitionLocalizationManager.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package org.flowable.cmmn.engine;
```
Comment (dbmalkovsky) [inline modules/flowable-cmmn-engine/src/main/java/org/flowable/cmmn/engine/CaseDefinitionLocalizationManager.java:1]: This class is missing the license header block.

File: modules/flowable-cmmn-engine/src/main/java/org/flowable/cmmn/engine/impl/history/HistoricPlanItemInstanceQueryImpl.java [hunk 3]
```diff
@@ -12,12 +12,9 @@
  */
 package org.flowable.cmmn.engine.impl.history;
 
-import java.util.Collection;
```
Comment (dbmalkovsky) [inline modules/flowable-cmmn-engine/src/main/java/org/flowable/cmmn/engine/impl/history/HistoricPlanItemInstanceQueryImpl.java:15]: same comment about import ordering.

File: modules/flowable-cmmn-engine/src/main/java/org/flowable/cmmn/engine/impl/persistence/entity/CaseDefinitionEntityImpl.java [hunk 4]
```diff
@@ -68,6 +74,9 @@ public String getKey() {
 
     @Override
     public String getDescription() {
+        if(StringUtils.isNotBlank(localizedDescription)) {
```
Comment (dbmalkovsky) [inline modules/flowable-cmmn-engine/src/main/java/org/flowable/cmmn/engine/impl/persistence/entity/CaseDefinitionEntityImpl.java:77]: inconsistent formatting; typically it is `if (` and not `if(`.

**Comment by DavidPLamas** [comment]
Applied changes as requested @dbmalkovsky.
I squashed my commit as I didn't know if you would want multiple commits for the same pull request. Hope it's ok.

**Comment by dbmalkovsky** [comment]
On Sept 6 Tijs did some engine configuration  refactoring in `HistoricCaseInstanceQueryImpl` so the `CmmnEngineConfiguration` is a class variable so you don't have to look it up. 

Line 50: `protected CmmnEngineConfiguration cmmnEngineConfiguration;`

And it is set (see lines 92-100) where the configuration is passed into the constructor.:

File: modules/flowable-cmmn-engine/src/main/java/org/flowable/cmmn/engine/impl/persistence/entity/PlanItemInstanceEntityImpl.java [hunk 5]
```diff
@@ -213,6 +214,9 @@ public void setPlanItemDefinitionType(String planItemDefinitionType) {
     }
     @Override
     public String getName() {
+        if (StringUtils.isNotBlank(localizedName)) {
```
Comment (dbmalkovsky) [inline modules/flowable-cmmn-engine/src/main/java/org/flowable/cmmn/engine/impl/persistence/entity/PlanItemInstanceEntityImpl.java:217]: There is no import for `StringUtils`.

**Comment by DavidPLamas** [comment]
Hi @dbmalkovsky,

I initially fixed the merge conflicts in GitHub as didn't notice other things would also need to be changed.
I've now fixed the things you pointed out and everything should be fine by now.

**Comment by DavidPLamas** [comment]
Hi, 

I just noticed Flowable 6.6 has been released. Does it mean this pull request isn't going to be included or can it be included later?

**Comment by tijsrademakers** [comment]
Hi @DavidPLamas , yesterday 6.6.0 was released indeed. We didn't have time to fully look into this PR yet, but we want to do a 6.6.1 release next and then we can look into this PR as well.

**Comment by DavidPLamas** [comment]
Hi @tijsrademakers, are there any updates on this issue? Are there any plans to merge with the actual code?

**Comment by tijsrademakers** [comment]
Hi @DavidPLamas, thank you for the reminder. We are working towards a 6.6.1 release in the next period and we will come back to this PR for that.

File: modules/flowable-cmmn-engine/src/main/java/org/flowable/cmmn/engine/impl/persistence/entity/CaseInstanceEntityImpl.java [hunk 6]
```diff
@@ -129,6 +131,9 @@ public void setBusinessKey(String businessKey) {
     }
     @Override
     public String getName() {
+        if(StringUtils.isNotBlank(localizedName)) {
```
Comment (dbmalkovsky) [inline modules/flowable-cmmn-engine/src/main/java/org/flowable/cmmn/engine/impl/persistence/entity/CaseInstanceEntityImpl.java:134]: This is very minor but there should be a space between the `if` and `(`

**Comment by tijsrademakers** [comment]
Thanks for the quick changes yesterday @DavidPLamas

**Pull request merged**